devito/nodes.py

```diff
@@ -592,3 +633,67 @@ def __init__(self, expr, dtype):
     def ccode(self):
         ctype = c.dtype_to_ctype(self.dtype)
         return c.Initializer(c.Value(ctype, ccode(self.expr.lhs)), ccode(self.expr.rhs))
+
+
+# Iteration utilities
+
+class IterationProperty(object):
+
+    """
+    An IterationProperty is an object that can be used to decorate an Iteration.
+    """
+
+    def __init__(self, name, val=None):
+        self.name = name
+        self.val = val
+
+    def __eq__(self, other):
+        if not isinstance(other, IterationProperty):
+            return False
+        return self.name == other.name and self.val == other.val
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __str__(self):
+        return self.name if self.val is None else '%s%s' % (self.name, str(self.val))
+
+    def __repr__(self):
+        if self.val is None:
+            return "Property: %s" % self.name
+        else:
+            return "Property: %s[%s]" % (self.name, str(self.val))
+
+
+SEQUENTIAL = IterationProperty('sequential')
+PARALLEL = IterationProperty('parallel')
+VECTOR = IterationProperty('vector-dim')
+ELEMENTAL = IterationProperty('elemental')
+REMAINDER = IterationProperty('remainder')
+
+known_properties = [SEQUENTIAL, PARALLEL, VECTOR, ELEMENTAL, REMAINDER]
+
+
+def tagger(i):
+    handle = IterationProperty('tag', i)
+    if handle not in known_properties:
+        known_properties.append(handle)
+    return handle
+
+
+def ntags():
+    return len(known_properties) - ntags.n_original_properties
+ntags.n_original_properties = len(known_properties)  # noqa
+
+
+class UnboundedIndex(object):
+
+    """
+    A generic loop iteration index used to traverse an iteration space in a
+    non-linear way (e.g., non-unitary increment).
```

Comment:
Since `Iteration` has a built-in way to have non-unitary increments, ~I think~ having that as the example here is confusing ~at the very least~ if not wrong (non-unitary != non-linear).